### ISSUE TITLE
False positive for `Style/RedundantLineContinuation`

### ISSUE THREAD

**Issue by marocchino** [rubocop/rubocop#12797]
I found more edge case for this lint.

--------

## Expected behavior

No offence.

## Actual behavior

```
$ ruby -e 'puts "a = b \\\n   || c"' | bundle exec rubocop --stdin test.rb --only Style/RedundantLineContinuation
Inspecting 1 file
C

Offenses:

test.rb:1:7: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
a = b \ ...
      ^

$ ruby -e 'puts "a = b \\\n  if c"' | bundle exec rubocop --stdin test.rb --only Style/RedundantLineContinuation
Inspecting 1 file
C

Offenses:

test.rb:1:7: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
a = b \ ...
      ^

$ ruby -e 'puts "a = b if c \\\n ||  d"' | bundle exec rubocop --stdin test.rb --only Style/RedundantLineContinuation
Inspecting 1 file
C

Offenses:

test.rb:1:12: C: [Correctable] Style/RedundantLineContinuation: Redundant line continuation.
a = b if c \ ...
           ^
```

Note that 1st, 2nd one maybe duplicate of https://github.com/rubocop/rubocop/issues/12793 

## Steps to reproduce the problem

see Actual behavior

## RuboCop version

```
$ rubocop -V
1.62.1 (using Parser 3.3.0.5, rubocop-ast 1.31.2, running on ruby 3.1.3) [arm64-darwin22]
  - rubocop-capybara 2.20.0
  - rubocop-factory_bot 2.25.1
  - rubocop-performance 1.20.2
  - rubocop-rails 2.24.0
  - rubocop-rspec 2.27.1
```

**Comment by koic** [comment]
I've opened #12798. The case regarding `if` is likely to be fixed in #12793.
